### PR TITLE
feat: add r/demo/users

### PR DESCRIPTION
**Pull request by DIGIX666**
Hi @leohhhn ,

I added `r/demo/users`  to  `p/memeland`.

I was able to test it locally, but I wanted to do it with the `Adena` version as well. However, I couldn't find a way to have my Adena public key in my `gnokey list`, and from what I found, I believe it's not possible.
I just need to do a unit test on it to be complete. If you could review it when you have time, that would be great. Thank you!

SCR-20240625-pkhc
SCR-20240625-pkpn
SCR-20240625-pksb
SCR-20240625-pleu

### CODE REVIEW

**Comment by leohhhn** [comment]
Hey, can you change the panic to be more descriptive? Something like "you need to register in r/demo/users to be able to post" - or similar.

**Comment by DIGIX666** [comment]
> Hey, can you change the panic to be more descriptive? Something like "you need to register in r/demo/users to be able to post" - or similar.

For sure, this is a new `panic` message https://github.com/gnolang/memeland/pull/24/commits/bfaa99f58393e2c930c55cdb36a622885b4f7f60

File: api/p/memeland/memeland.gno [hunk 1]
```diff
@@ -38,19 +39,30 @@ func NewMemeland() *Memeland {
...
 // PostMeme - Adds a new post
 func (m *Memeland) PostMeme(data string, timestamp int64) string {
 	if data == "" || timestamp <= 0 {
 		panic("timestamp or data cannot be empty")
 	}
 
+	// check address is register
+	author := std.PrevRealm().Addr()
+	if !isUserRegistered(author) {
+		panic("address not registered. Can you go to register in r/demo/users for post after")
```
Comment (leohhhn) [inline api/p/memeland/memeland.gno:56]:
```suggestion
		panic("you have to register in r/demo/users to post!")
```

**Review by leohhhn — APPROVED** [review]
Approving, just suggested a better panic message :)

File: api/p/memeland/memeland.gno (hunk 1 above)
Comment (DIGIX666) [reply api/p/memeland/memeland.gno]: Thanks @leohhhn, I'll be more careful on my next `panic`

**Pull request merged**